### ISSUE TITLE
NoneType value parameter using EMR filters

### ISSUE THREAD

**Issue by aiell0** [capitalone/cloud-custodian#453]
Hello,

I am using the following filter for finding EMR resources in our AWS cloud:

```
 - type: value
    value_type: age
    key: "Status.Timeline.ReadyDateTime"
    value: 7
    op: less-than
```

In some select cases, EMR clusters will not have a `ReadyDateTime`, and this causes the parser in the filtering module to fail at line 338. The following fix has sufficed for me thus far:

```
            try:
                value = parse(value)
            except AttributeError:
                value = 0
```

Upon fixing that, another crash occurred in the operator module due to the same underlying problem, at line 308. The following fix patches that up:

```
       try:
            return op(r, v)
        except TypeError:
            return False
```

**Comment by kapilt** [comment]
Thanks I noticed some issues here recently around null values. I'm thinking we shouldn't default a value unless it's been expressly set in the policy, else treat it as a not match.  I'll take a look at cleaning this up.

**Comment by kapilt** [comment]
also related #468

**Comment by kapilt** [comment]
closed in #490
